dask_cuda/utils.py

```diff
@@ -357,3 +362,38 @@ def wait_workers(
             return False
         else:
             time.sleep(0.1)
+
+
+async def _all_to_all(client, cleanup=True):
+    """
+    Trigger all to all communication between workers and scheduler
+    """
+    workers = list(client.scheduler_info()["workers"])
+    futs = []
+    for w in workers:
+        bit_of_data = b"0" * 1
+        data = client.map(lambda x: bit_of_data, range(1), pure=False, workers=[w])
+        futs.append(data[0])
+
+    await wait(futs)
+
+    def f(x, y):
+        pass
+
+    new_futs = []
+    for w in workers:
+        for fut1, fut2 in combinations(futs, 2):
+            data = client.submit(f, fut1, fut2, workers=[w], pure=False)
+            new_futs.append(data)
```

Comment:
What this seems to be doing is: for each worker `A`, for each pair of workers `(B,C)`, pull data from `B` and `C` and combine them on `A`, i.e. `O(n_workers^3)` operations. If the goal is simply to "prime" each worker-to-worker connection with a small message, then I believe it would be sufficient to do the following, which only does `O(n_workers^2)` operations:

```
def f(x):
    pass

new_futs = []
for w in workers:
    for data_in in futs:
        data_out = client.submit(f, data_in, workers=[w], pure=False)
        new_futs.append(data_out)
```

Reply:
Addressed in 381f863 .  Thanks @manopapad